app/components/avo/views/resource_index_component.rb

```diff
@@ -29,6 +30,28 @@ def title
     end
   end
 
+  def view_type
+    @index_params[:view_type]
+  end
+
+  def bulk_edit_path
+    # Add the `view` param to let Avo know where to redirect back when the user clicks the `Cancel` button.
+    args = {via_view: "index"}
+
+    if @parent_record.present?
+      args = {
+        via_resource_class: parent_resource.class.to_s,
+        via_record_id: @parent_record.to_param
+      }
+    end
+
+    helpers.edit_bulk_update_path(resource: @resource, **args)
+  end
+
+  def available_view_types
+    @index_params[:available_view_types]
+  end
```

Comment:
```suggestion
```

This method was refactored and can be accessed from the resource: `resource.avialable_view_types`